src/services/safes.rs

```diff
@@ -18,7 +19,12 @@ pub async fn get_safe_info_ex(
 ) -> ApiResult<SafeState> {
     let info_provider = DefaultInfoProvider::new(chain_id, context);
     let safe_info = info_provider.safe_info(safe_address).await?;
-    let safe_info_ex = safe_info.to_safe_info_ex(&info_provider).await;
+    let supported_master_copies = get_master_copies(&context, chain_id)
```

Comment:
We should not do this. As you mentioned above `get_master_copies` has a separate info provider and therefore it will not be reusing the in-memory cache, which here would be helpful.

Reply:
I think the only option here would be to duplicate some code. As I need access to the `client` in `Context` to issue the request.

Reply:
the fetching of the master copies could be part of the info provider interface.

Reply:
I thought of it too, but I am starting to lose sight as to what is the purpose of the `info_provider`

Reply:
maybe the `master_copies` do constitute a piece of information that can be considered for "enriching" other responses (namely `SafeInfoEx`) and it wouldn't be the first time we add a method we would return directly from the `info_provider`

Reply:
I would still think of a way to refactor the `info_provider` into more specific subcomponents, as it's starting to look like a monolith

Reply:
I would still think of a way to refactor the `info_provider` into more specific subcomponents, as it's starting to look like a monolith

Reply:
I still think the current approach feels weird. Having a method that just perform the request. And that method is also in the module of another request handler ... idk ... I would prefer it to be part of the info provider or as you mentioned and extension. but we alreay pull safe info and token info from there, so with our current approach it would fit. And then in a next step these method could be extracted to a separate entity